### PR TITLE
Fetch areas relations

### PR DESCRIPTION
**Pull request by Marianasilvago**
### What

Fetch relationships from rds based area code and clean up mongo dependency

### How to review
Make sure tests pass.

### Who can review

Anyone.

### CODE REVIEW

File: service/initialise.go [hunk 1]
```diff
@@ -81,13 +83,34 @@ func (e *ExternalServiceList) GetPGXPool(ctx context.Context, cfg *config.Config
 	return pgxConn, nil
 }
 
+func (e *ExternalServiceList) getRDSDB(ctx context.Context, cfg *config.Config) (api.RDSAreaStore, error) {
+	rds, err := e.Init.DoGetRDSDB(ctx, cfg)
+	if err != nil {
+		log.Error(ctx, "failed to create rds client", err)
```
Comment (ChrysmOre) [inline service/initialise.go:89]: Please can we stop logging errors and then returning them! You end up with the same error logged multiple times, not having an accurate picture of what's going on. Much better to return the error wrapped with the message `return nil, fmt.Errorf("failed to create rds client: %w", err)`

File: service/service.go [hunk 2]
```diff
@@ -43,7 +44,14 @@ func Run(ctx context.Context, cfg *config.Config, serviceList *ExternalServiceLi
 	if err != nil {
 		log.Fatal(ctx, "failed to initialise mongo db client", err)
 		return nil, err
-	} 
+	}
+
+	// Get RDS client
+	rds, err := serviceList.getRDSDB(ctx, cfg)
```
Comment (mike-kelly-methods) [inline service/service.go:50]: We're replicating work here - a new PGX Pool is being opened at line `57` below - complete with external service list initialisation also. See here: https://github.com/ONSdigital/dp-areas-api/blob/6a6a7a395422eb9ec790b9deaba2dd24dcfa3220/pgx/pgx.go#L33

File: service/service.go [hunk 3]
```diff
@@ -53,7 +61,7 @@ func Run(ctx context.Context, cfg *config.Config, serviceList *ExternalServiceLi
 	}
 
 	// Setup the API
-	a, _ := api.Setup(ctx, cfg, r, mongoDB, pgxConn)
+	a, _ := api.Setup(ctx, cfg, r, mongoDB, pgxConn, rds)
```
Comment (mike-kelly-methods) [inline service/service.go:64]: As a result, we're passing both pools to the api setup method.

File: rds/rds.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+func (r *RDS) ValidateArea(areaCode string) error {
+	area := "select code from area where code = $1"
+	var code string
+	err := r.conn.QueryRow(context.Background(), area, areaCode).Scan(&code)
+	return err
+}
+
+func (r *RDS) GetRelationships(areaCode string) ([]*models.AreaBasicData, error) {
+	var relationships []*models.AreaBasicData
+	relationshipQuery := "select code, name from area, area_relationship where rel_area_code = code and area_code = $1"
```
Comment (mike-kelly-methods) [inline rds/rds.go:53]: We should have a package for SQL queries, and call them in the method that they're utilised in; a possible location could be:

`github.com/mikek/dp-areas-api/sql/queries.go`

`var GetRelationshipsQuery = "select code, name from area, area_relationship where rel_area_code = code and area_code = $1"`

`relationshipQuery := sql.GetRelationshipsQuery`

File: api/area_test.go [hunk 5]
```diff
@@ -45,26 +40,26 @@ func dbArea(id string, name string) *models.Area {
...
 	So(err, ShouldBeNil)
 	cfg.DefaultLimit = 0
 	cfg.DefaultOffset = 0
-	return api.Setup(context.Background(), cfg, mux.NewRouter(), mockedAreaStore, nil)
+	return api.Setup(context.Background(), cfg, mux.NewRouter(), mockedAreaStore, &mock.RDSAreaStoreMock{})
 }
 
-// GetAPIWithPGXMocks returns mocked API with PGX mock attached
-func GetAPIWithPGXMocks(mockedAreaStore api.AreaStore, pgxMock *pgxTest.PGX) (*api.API, error) {
+// GetAPIWithRDSMocks also used in other tests, so exported
```
Comment (mike-kelly-methods) [inline api/area_test.go:54]: Only used locally in this test package, so don't need to export

**Review by mike-kelly-methods — APPROVED** [review]
👍

**Review by mike-kelly-methods — APPROVED** [review]
👍

**Pull request merged**